test/powershell/Modules/Microsoft.PowerShell.Utility/Remove-Alias.Tests.ps1

```diff
@@ -0,0 +1,45 @@
+Describe "Remove-Alias" -Tags "CI" {
+    It "Remove-Alias should remove a non-readonly alias"{
+        {
+            Set-Alias -Name "foo" -Value "bar"
+            Remove-Alias -Name "foo"
+            Get-Alias -Name "foo" -ErrorAction Stop
+        } | ShouldBeErrorId 'ItemNotFoundException,Microsoft.PowerShell.Commands.GetAliasCommand'
+    }
+
+    It "Remove-Alias should throw on a readonly alias"{
+        {
+            Set-Alias -Name "foo" -Value "bar" -Option ReadOnly
+            Remove-Alias -Name "foo" -ErrorAction Stop
+        } | ShouldBeErrorId 'AliasNotRemovable,Microsoft.PowerShell.Commands.RemoveAliasCommand'
+    }
+
+    It "Remove-Alias should remove a non-readonly alias with force"{
+        {
+            Set-Alias -Name "foo" -Value "bar"
+            Remove-Alias -Name "foo" -Force
+            Get-Alias -Name "foo" -ErrorAction Stop
+        } | ShouldBeErrorId 'ItemNotFoundException,Microsoft.PowerShell.Commands.GetAliasCommand'
+    }
+
+    It "Remove-Alias should remove a readonly alias with force"{
+        {
+            Set-Alias -Name "foo" -Value "bar" -Option ReadOnly
+            Remove-Alias -Name "foo" -Force
+            Get-Alias -Name "foo" -ErrorAction Stop
+        } | ShouldBeErrorId 'ItemNotFoundException,Microsoft.PowerShell.Commands.GetAliasCommand'
+    }
+
+    It "Remove-Alias should throw if alias does not exist"{
+        {            
```

Comment:
I think we should protect themselves and add `Get-Alias -Name "foo" | Should BeNullorEmpty` before trying to remove.

Reply:
Do you mean as a separate tests before the "Remove-Alias should throw if alias does not exist" test or just like this?
```Powershell
It "Remove-Alias should throw if alias does not exist"{
        {
            Get-Alias -Name "foo" | Should BeNullorEmpty
            Remove-Alias -Name "foo" -ErrorAction Stop            
        } | ShouldBeErrorId 'ItemNotFoundException,Microsoft.PowerShell.Commands.RemoveAliasCommand'
    }
```

Reply:
Yes.